File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
    int count = 0;
    // count the number of val occurrences
    for (auto i = 0; i < nums.size(); i++) {
        if (nums[i] == val) {
            count++;
        }
        else {
            nums[i - count] = nums[i]; // shift by number of val occurrences so far
        }
    }
    int k = (int)nums.size() - count;
    return k;  
    }
};

File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        
        output = [-1] * (n + 1)

        for i in range(n + 1):
            count = 0
            j = i

            while j > 0:
                if j & 1 == 1:
                    count += 1
                j = j >> 1
            
            output[i] = count

        return output
